### PR TITLE
Allow assignment selection from timer

### PR DESCRIPTION
**Pull request by jrbotros**

### CODE REVIEW

File: orchestra/static/orchestra/timing/timecard/controller.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 (function() {
-  'use strict';
+  // 'use strict';
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timecard/controller.js]: uncomment

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 2]
```diff
@@ -40,6 +41,17 @@
...
                 workTimer.timerRunning = true;
                 workTimer.description = response.data.description;
+                if (Object.keys(orchestraTasks.tasksByAssignmentId)) {
+                  orchestraTasks.data.then(function() {
+                    workTimer.task = orchestraTasks.tasksByAssignmentId[response.data.assignment];
+                  });
+                }
+                else {
+                  $rootScope.$on('orchestraTasks:update', function() {
+                    console.log('asdgdgsa');
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js]: remove

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 3]
```diff
@@ -40,6 +41,17 @@
                 workTimer.toggledOn = true;
                 workTimer.timerRunning = true;
                 workTimer.description = response.data.description;
+                if (Object.keys(orchestraTasks.tasksByAssignmentId)) {
+                  orchestraTasks.data.then(function() {
+                    workTimer.task = orchestraTasks.tasksByAssignmentId[response.data.assignment];
+                  });
+                }
+                else {
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js]: talked offline about removing this else section

File: orchestra/static/orchestra/timing/timer/directives.js [hunk 4]
```diff
@@ -77,21 +91,35 @@
...
+              assignment: assignment_id,
+              description: workTimer.description
+            };
+            $http.post(url, data)
             .catch(function() {
                 alert('Could not update timer description');
             });
           };
 
+          $scope.$watch('workTimer.task', function(newVal, oldVal) {
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timer/directives.js:109]: do you need this watch?

File: orchestra/static/orchestra/timing/timecard/controller.js [hunk 5]
```diff
@@ -52,20 +52,14 @@
         return {
           description: entry.description,
           timeWorked: entry.time_worked.componentize(),
-          assignment: orchestraTasks.tasksById[entry.assignment]
+          task: orchestraTasks.tasksByAssignmentId[entry.assignment]
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timecard/controller.js:54]: is there a race condition here? Where the promise hasn't resolved yet, but entry.assignment is loaded from time entries?

File: orchestra/static/orchestra/timing/timecard/controller.js (hunk 1 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js]: ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 2 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js]: ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 3 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js]: ✅

File: orchestra/static/orchestra/timing/timer/directives.js (hunk 4 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js:109]: Yes—this is what updates the task server side when a new task is selected in the interface

Comment (jrbotros) [inline orchestra/static/orchestra/timing/timer/directives.js:109]: ✅

File: orchestra/static/orchestra/timing/timecard/controller.js (hunk 5 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js:54]: Yeah, but only if the client opens up the edit view before the tasks have loaded from the server—I think this is probably a pretty small case, but I'll fix it to be sure.

Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js:54]: ✅

**Pull request merged**